### PR TITLE
Cmc whitelist

### PR DESCRIPTION
**Pull request by rmisio**
Adjusts the crypto data script to pull the whitelist from the ticker instead of hard-coding it.

### CODE REVIEW

File: bin/getCryptoData.js [hunk 1]
```diff
@@ -83,76 +92,88 @@ function getIcon(coin) {
...
+
+          const callGetIcon = (index = 0) => {
+            getIcon(results.data[index]);
+            if (index + 1 < results.data.length) {
+              // Give some space between requests, otherwise kicking off so many at once
+              // leads to many of them failing.
+              setTimeout(() => callGetIcon(index + 1), 50);
+            }
+          };
+          callGetIcon();
```
Comment (jjeffryes) [inline bin/getCryptoData.js]: Will this fire if results.data turns out to be empty, and attempt to call getIcon on undefined? Should it have a if (results.data.length) check?

File: bin/getCryptoData.js [hunk 2]
```diff
@@ -83,76 +92,88 @@ function getIcon(coin) {
...
+
+      getCoinList()
+        .then(results => {
+          // remove duplicates from the data set using our whitelist
+          const newList = [];
+          results.data.forEach((c, i) => {
+            const furtherResults = results.data.slice(i + 1);
+            if (
+              !(whitelist[c.symbol] && whitelist[c.symbol] !== c.id) &&
+              !furtherResults.find(coin => coin.symbol === c.symbol)
```
[the remark this answers is not in the document]
Comment (jjeffryes) [reply bin/getCryptoData.js]: For example, when I ran the script, DraftCoin, id 1120, was replaced by DigiFinexToken, id 2878.

[the remark this answers is not in the document]
Comment (rmisio) [reply bin/getCryptoData.js]: @jjeffryes Where are you seeing DigiFinexToken coming in? I'm seeing DraftCoin being used as both the icon and the name:
https://github.com/OpenBazaar/openbazaar-desktop/blob/master/imgs/cryptoIcons/DFT-icon.png
https://github.com/OpenBazaar/openbazaar-desktop/blob/master/js/languages/en_US.json#L2880

**Comment by rmisio** [comment]
@jjeffryes this is ready for re-review.

File: bin/getCryptoData.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rmisio) [reply bin/getCryptoData.js]: In any case, looking at the code, it does look like the last one would be used which is not what I intended, so I tweaked it.

**Pull request merged**